Chained.h

```diff
@@ -11,197 +13,195 @@
-            std::cout << "Mean Collisions: " << collisionStats->mean << "\n";
-            std::cout << "Stdev Collisions: " << collisionStats->stdev << "\n";
+            Statistic collisionStats = evaluateStats(&collisions);
+            std::cout << "Total Collisions: " << collisionStats.total << "\n";
+            std::cout << "Mean Collisions: " << collisionStats.mean << "\n";
+            std::cout << "Stdev Collisions: " << collisionStats.stdev << "\n";
 
-            std::cout << "Total Number of Lists: " << hashmapSize << "\n";
+            std::cout << "Total Number of Lists: " << hashmap_.size() << "\n";
             std::cout << "Number of Lists With Zero Elements: " << numWith0 << "\n";
             std::cout << "Number of Lists with One Element: " << numWith1 << "\n";
             std::cout << "Number of Lists with more than One Element: " << numWithMore << "\n";
-            //I could make a display of all element sizes. Box plot? Scatterplot?
+            //Histogram?
 
             std::cout << "Number of elements in the biggest List: " << elements[elements.size()-1] << "\n";
             std::cout << "Number of elements in the smallest List: " << elements[0] << "\n";
         }
+        std::cout << std::endl;
     }
-};
 
-template<>
-uint64_t ChainedMap<std::string, uint64_t>::defaultGetReturn() {
-    return -1;
-}
-template<>
-Product* ChainedMap<uint64_t, Product*>::defaultGetReturn(){
-    return nullptr;
-}
+    double getLoadFactor() override {
+        return (static_cast<float>(getNumPairs()))/(static_cast<float>(hashmap_.size()));
+    }
+
+    uint64_t getFullSize() override {
+        return hashmap_.size();
+    }
+
+    uint64_t getNumPairs() override {
+        return numPairs_;
+    }
+
+private:
+    //default max load factor before doubling size is 0.75
+    double MAX_LOAD_FACTOR_ = 0.75;
+
+    struct KeyValuePair{
+    public:
+        KeyValuePair(Key k, Value v) : key_(k), value_(v) {}
+        Key key_;
+        Value value_;
+    };
+
+    std::vector<std::list<KeyValuePair>> hashmap_;
+
+    uint64_t numPairs_ = 0;
 
+    std::hash<Key> hasher_;
+
+    uint64_t hash(Key k, uint64_t m){
+        return hasher_(k) % m;
+    }
+
+    void doubleSize(){
+        std::vector<std::list<KeyValuePair>> newMap(hashmap_.size() * 2);
+        for(const auto& bucket : hashmap_){
+            for(const auto& kvp : bucket){
+                uint64_t index = hash(kvp.key_, hashmap_.size() * 2);
+                newMap[index].push_back(kvp);
+            }
+        }
+        hashmap_ = newMap;
+    }
+
+    void printKeyValuePair(KeyValuePair kvp){
+        std::cout << " Invalid types for printing ";
+    }
+
+};
 
 template<>
-uint64_t ChainedMap<std::string, uint64_t>::hash(std::string k){
-    return hasher(k)%hashmapSize;
+uint64_t ChainedMap<uint64_t, Product*>::hash(uint64_t k, uint64_t m) {
+    uint64_t out[2];
+    MurmurHash3_x64_128(&k, sizeof(k), 123, out);
+    return out[1] % m;
```

Comment:
I mean, this is fine.
But if there is a 64bit interface it is better to use that.
Because it looks a little weird to ask teh Murmur library for 128bits of hash material, and then throw away the bottom 64-bits.  Hashing is expensive.  Don't ask for hash bits you don't need.  Same with randomness.

The only acceptable one letter variable names are i, j, k for loop induction variables.  That is all.  Sometimes `char c;` for chracter-at-a-time IO.